composites/Plugin/ContentAnalysis/components/ContentAnalysis.js

```diff
@@ -1,17 +1,218 @@
+			case "OK":
+				return colors.$color_ok;
+			case "bad":
+				return colors.$color_bad;
+			default:
+				return colors.$color_score_icon;
+		}
+	}
+
+	/**
+	 * Returns an AnalysisResult component for each result.
+	 *
+	 * @param {array} results The analysis results.
+	 *
+	 * @returns {array} A list of AnalysisResult components.
+	 */
+	getResults( results ) {
+		return results.map( ( result ) => {
+			let color = this.getColor( result.rating );
+			return <AnalysisResult
+				key={ result.id }
+				text={ result.text }
+				bulletColor={ color }
+				hasMarksButton={ result.hasMarks }
+				ariaLabel={ this.props.intl.formatMessage( messages.highlight ) }
+				pressed={ result.id === this.state.checked }
+				buttonId={ result.id }
+				onButtonClick={ this.handleClick.bind( this, result.id ) }
+			/>;
+		} );
+	}
+
+	/**
+	 * Renders a ContentAnalysis component.
+	 *
+	 * @returns {ReactElement} The rendered ContentAnalysis component.
+	 */
+	render() {
+		let problemsResults = this.props.problemsResults;
+		let improvementsResults = this.props.improvementsResults;
+		let goodResults = this.props.goodResults;
+		let considerationsResults = this.props.considerationsResults;
+		let errorsResults = this.props.errorsResults;
+
+		// Analysis collapsibles are only rendered when there is at least one analysis result for that category present.
+		return (
+			<ContentAnalysisContainer>
+				<LanguageNotice>
+					<FormattedMessage
+						id="content-analysis.language-notice"
+						defaultMessage="Your site language is set to {language}."
+						values={ { language: <strong>{ this.props.language }</strong> } } />
+					<ChangeLanguageLink href={ this.props.changeLanguageLink }>
+						{ this.props.intl.formatMessage( messages.languageNoticeLink ) }
+					</ChangeLanguageLink>
+				</LanguageNotice>
+				{ errorsResults.length > 0
+					? <AnalysisCollapsible initialIsOpen={ true } title={ this.props.intl.formatMessage( messages.errorsHeader ) }>
+						{ this.getResults( errorsResults ) }
+					</AnalysisCollapsible>
+					: null }
+				{ problemsResults.length > 0
+					? <AnalysisCollapsible initialIsOpen={ true } title={ this.props.intl.formatMessage( messages.problemsHeader ) }>
+						{ this.getResults( problemsResults ) }
+					</AnalysisCollapsible>
+					: null }
+				{ improvementsResults.length > 0
+					? <AnalysisCollapsible title={ this.props.intl.formatMessage( messages.improvementsHeader ) }>
+						{ this.getResults( improvementsResults ) }
+					</AnalysisCollapsible>
+					: null }
+				{ considerationsResults.length > 0
+					? <AnalysisCollapsible title={ this.props.intl.formatMessage( messages.considerationsHeader ) }>
+						{ this.getResults( considerationsResults ) }
+					</AnalysisCollapsible>
+					: null }
+				{ goodResults.length > 0
+					? <AnalysisCollapsible title={this.props.intl.formatMessage( messages.goodHeader ) }>
+						{ this.getResults( goodResults ) }
+					</AnalysisCollapsible>
```

Comment:
If I'm not wrong, instead of these 5 ternaries:
condition ? component : null
we could simplify and just do:
condition && component